tests/cli/main_tool.py

```diff
@@ -36,3 +38,28 @@ def testToolWithArbitraryRecipe(self):
     tool.state.LogExecutionPlan()
     for recipe in tool._recipes_manager.GetRecipes():
       tool._recipes_manager.DeregisterRecipe(recipe)
+
+  def testRecipeSetupArgs(self):
+    """Checks that all recipes pass the correct arguments to their modules."""
+    tool = dftimewolf_recipes.DFTimewolfTool()
+    tool.LoadConfiguration()
+    tool.ReadRecipes()
+
+    # We want to access the tool's sate object to load recipes and go through
+    # modules.
+    # pylint: disable=protected-access
+    tool._state =  dftw_state.DFTimewolfState(config.Config)
+
+    for recipe in tool._recipes_manager.GetRecipes():
+      tool._state.LoadRecipe(recipe.contents, dftimewolf_recipes.MODULES)
+      for module in recipe.contents['modules']:
+        runtime_name = module.get('runtime_name', module['name'])
+        setup_func = tool.state._module_pool[runtime_name].SetUp
+        expected_args = set(inspect.getfullargspec(setup_func).args)
+        expected_args.remove('self')
+        provided_args = set(module['args'])
+
+        self.assertEqual(
```

Comment:
Do we need to be concerned by argument order?
Say, I have a recipe that takes `--arg1` and `--arg2`. Due to `--`, both are optional. If I pass `dftimewolf myrecipe --arg1=foo --arg2=bar` or `dftimewolf myrecipe --arg2=bar --arg1=foo` will it work?

Reply:
They all end up being passed as kwargs, so the order doesn't really matter so long as the names match.

Now that all recipes are passing all parameters (optional or not), a good question to ask is whether we still want positional vs. keyword parameters in the module's SetUp() functions, or if we should just consistently have them be positional, and specify the default value in the recipe. (I would favor the latter)

Reply:
The latter SGTM